sandbox/cgroup/monitor.go

```diff
@@ -0,0 +1,159 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package cgroup
+
+import (
+	"errors"
+	"path"
+	"sync"
+
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/osutil/inotify"
+)
+
+type inotifyWatcher struct {
+	wd           *inotify.Watcher
+	doOnce       sync.Once
+	addWatchChan chan *groupToWatch
+	// groupList and pathList are accessed only from inside the watcherMainLoop,
+	// so no locks are needed.
+	groupList []*groupToWatch
+	pathList  map[string]int32
+}
+
+type groupToWatch struct {
+	name    string
+	folders []string
+	channel chan string
+}
+
+var currentWatcher *inotifyWatcher = &inotifyWatcher{
+	wd:           nil,
+	pathList:     make(map[string]int32),
+	addWatchChan: make(chan *groupToWatch),
+}
+
+func (this *inotifyWatcher) addWatch(newWatch *groupToWatch) {
```

Comment:
We don't use 'this' naming for the instance itself, instead  we use abbreviations, same goes for all the functions in this file. 
```suggestion
func (iw *inotifyWatcher) addWatch(newWatch *groupToWatch) {
```

Reply:
Done.